source/simulator/checkpoint_restart.cc

```diff
@@ -610,7 +610,27 @@ namespace aspect
 
     ar &statistics;
 
+    // Serialize various plugin systems. In many cases, plugins are stateless and
+    // serialization will not do anything. But some are stateful (for example the
+    // dynamic core boundary temperature plugin) and need to be serialized.
+    ar &mesh_refinement_manager;
+    ar &heating_model_manager;
     ar &postprocess_manager;
+    ar &boundary_temperature_manager;
+    ar &boundary_convective_heating_manager;
+    ar &boundary_composition_manager;
+    ar &prescribed_solution_manager;
+    ar &boundary_velocity_manager;
+    ar &boundary_traction_manager;
+
+// The following two are not manager classes but straight up plugins and so don't
+// currently have the ability to serialize themselves. We should add those later.
+//    ar &prescribed_stokes_solution;
+//    ar &boundary_heat_flux;
```

Comment:
I know how to do that, but it requires a bit more work. So I'll leave that for a separate PR in this series.